knowledge_graphs/hardware_optimized_i7_13700k.py

```
@@ -0,0 +1,407 @@
+        memory = psutil.virtual_memory()
+        cpu_percent = psutil.cpu_percent(interval=1, percpu=True)
+        
+        # Per-core CPU usage (important for hybrid architecture)
+        p_core_usage = cpu_percent[:16]  # P-cores + hyperthreading
+        e_core_usage = cpu_percent[16:24] if len(cpu_percent) >= 24 else []
+        
+        # Thermal information (if available)
+        try:
+            temps = psutil.sensors_temperatures()
+            cpu_temp = temps.get('coretemp', [{}])[0].get('current', 0)
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)